StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1504AllAccessorsMustBeSingleLineOrMultiLine.cs

```diff
@@ -72,18 +76,50 @@ public class SA1504AllAccessorsMustBeSingleLineOrMultiLine : DiagnosticAnalyzer
             ImmutableArray.Create(Descriptor);
 
         /// <inheritdoc/>
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => SupportedDiagnosticsValue;
+
+        /// <inheritdoc/>
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(this.HandlePropertyDeclaration, SyntaxKind.PropertyDeclaration, SyntaxKind.IndexerDeclaration, SyntaxKind.EventDeclaration);
+        }
+
+        private void HandlePropertyDeclaration(SyntaxNodeAnalysisContext context)
         {
-            get
+            var property = (BasePropertyDeclarationSyntax) context.Node;
+
+            if (property.AccessorList == null 
+                || property.AccessorList.IsMissing
+                || property.AccessorList.Accessors.Count != 2)
+            {
+                return;
+            }
+
+            if (!this.AreAllAccessorsInMultiLines(property.AccessorList.Accessors) 
+               && !this.AreAllAccessorsInOneLine(property.AccessorList.Accessors))
             {
-                return SupportedDiagnosticsValue;
+                context.ReportDiagnostic(Diagnostic.Create(Descriptor, property.AccessorList.Accessors[1].GetLocation()));
```

Comment:
:question: Does StyleCop report this warning on the property/event/indexer name, or on the first accessor?

:bulb: Consider reporting the warning on a single token instead of an entire block.